tests/drivers/pwm/pwm_to_gpio_loopback/src/test_pwm_to_gpio_loopback.c

```diff
@@ -0,0 +1,84 @@
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/ztest.h>
+
+#include "test_pwm_to_gpio_loopback.h"
+
+#define TEST_PWM_PERIOD_USEC 100
+#define TEST_PWM_PULSE_USEC  15
+
+#define NUMBER_OF_CYCLE_TO_CAPTURE 5
+
+static struct gpio_callback pwm_input_cb_data;
+static uint32_t high, low;
```

Comment:
Should it be volatile? atomic? It is modified from the test and the callback (interrupt?)